cookbooks/calico/recipes/general_compute.rb

```diff
@@ -3,6 +3,14 @@
 
 # Find the BGP neighbors, which is everyone except ourselves.
 bgp_neighbors = search(:node, "role:compute").select { |n| n[:ipaddress] != node[:ipaddress] }
+# Overwrite chef's IPv6 address assignment because it's wrong.
+bgp_neighbors.each do |neighbor|
+    addresses = neighbor[:network][:interfaces][:eth0][:addresses].keys.map(&:to_s)
+    global_ipv6 = addresses.select do |address|
+        address.start_with? '2620:'
```

Comment:
Unless I'm mistaken the 2620 address range is allotted by our DHCP server, so you can't just use this prefix match to determine which IP address to use since a different network may not use the same prefix.